src/Microsoft.Fx.Portability.MetadataReader/ReflectionMetadataDependencyInfo.cs

```diff
@@ -196,5 +197,33 @@ public InvalidPEAssemblyException(Exception inner)
                 : base("Not a valid assembly", inner)
             { }
         }
+
```

Comment:
I replaced everything with ImmutableCollections and then did some checks around performance. Looks like using ImmutableCollections here will decrease performance by about 40% - we're doing all the building initially, which is known to drag performance with them. I think we should stick with the concurrent collections for now.